core/src/banking_stage.rs

```diff
@@ -706,49 +701,30 @@ impl BankingStage {
         let mut rebuffered_packet_count = 0;
         let mut consumed_buffered_packets_count = 0;
         let mut proc_start = Measure::start("consume_buffered_process");
-        let reached_end_of_slot;
         let num_packets_to_process = unprocessed_transaction_storage.len();
 
-        let (bank_start, poh_recorder_lock_time) = measure!(
-            poh_recorder.read().unwrap().bank_start(),
-            "poh_recorder.read",
-        );
-        slot_metrics_tracker.increment_consume_buffered_packets_poh_recorder_lock_us(
-            poh_recorder_lock_time.as_us(),
+        let reached_end_of_slot = unprocessed_transaction_storage.process_packets(
+            bank_start.working_bank.clone(),
+            banking_stage_stats,
+            slot_metrics_tracker,
+            |packets_to_process, payload| {
+                Self::do_process_packets(
+                    bank_start,
+                    payload,
+                    recorder,
+                    transaction_status_sender,
+                    gossip_vote_sender,
+                    banking_stage_stats,
+                    qos_service,
+                    log_messages_bytes_limit,
+                    &mut consumed_buffered_packets_count,
+                    &mut rebuffered_packet_count,
+                    &test_fn,
+                    packets_to_process,
+                )
+            },
         );
 
-        if let Some(BankStart {
-            working_bank,
-            bank_creation_time,
-        }) = bank_start
-        {
-            reached_end_of_slot = unprocessed_transaction_storage.process_packets(
-                working_bank.clone(),
-                banking_stage_stats,
-                slot_metrics_tracker,
-                |packets_to_process, payload| {
-                    Self::do_process_packets(
-                        max_tx_ingestion_ns,
-                        &working_bank,
-                        &bank_creation_time,
-                        payload,
-                        recorder,
-                        transaction_status_sender,
-                        gossip_vote_sender,
-                        banking_stage_stats,
-                        qos_service,
-                        log_messages_bytes_limit,
-                        &mut consumed_buffered_packets_count,
-                        &mut rebuffered_packet_count,
-                        &test_fn,
-                        packets_to_process,
-                    )
-                },
-            );
-        } else {
-            reached_end_of_slot = true;
-        }
-
         if reached_end_of_slot {
             slot_metrics_tracker.set_end_of_slot_unprocessed_buffer_len(
```

Comment:
nit: GH won't let me comment on ln 733 comment of `// We've hit the end of this ...`, that comment can be removed?

Reply:
removed